Orange/widgets/settings.py

```diff
@@ -462,7 +463,11 @@ def initialize(self, instance, data=None):
     def _migrate_settings(self, settings):
         """Ask widget to migrate settings to the latest version."""
         if settings:
-            self.widget_class.migrate_settings(settings, settings.pop(VERSION_KEY, 0))
+            try:
+                self.widget_class.migrate_settings(
+                    settings, settings.pop(VERSION_KEY, 0))
+            except Exception:
+                sys.excepthook(*sys.exc_info())
```

Comment:
@janezd, when context migration fails, the offending context is removed. Should settings also be cleared if they cannot be migrated? (settings.clear())

Reply:
You are the only one who really knows the mechanics of settings. If they can't be migrated, they mustn't be used. If calling `settings.clear()` is the way to achieve this, then, yes, we should call `settings.clear()` here.